### DISCUSSION TITLE
need help to add cookies to url's

### DISCUSSION THREAD

**Discussion by Rahul9354** [DigitalPebble/storm-crawler#1121] [Q&A] [answered]
can anyone help me on how to add cookie to a url before crawling it. I have a crawler based on storm crawler which is being used to crawl news website url's. No w I want to add cookies to the url's for a news website with premium membership.

**Comment by rzo1** [comment, marked as answer]
You could generate the cookie externally prior to the crawling and specify it in the seed metadata using the key **set-cookie**. You'd need to add that key to metadata.transfer in your conf so that it gets transmitted to the outlinks and persisted to the storage and enable ` http.use.cookies`.
